### PR TITLE
Updated service version to 7.3 and re-recorded tests for Key Vault libraries

### PR DESCRIPTION
**Pull request by vcolin7**
# Description

Updated service version to 7.3 and re-recorded tests for Key Vault libraries. Also re-generated Key Vault Administration based on this swagger updates included in [this PR](https://github.com/Azure/azure-rest-api-specs/pull/18028).

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-administration/swagger/autorest.md [hunk 1]
```diff
@@ -52,14 +52,14 @@ context-client-method-parameter: true
...
+input-file: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.3/rbac.json
 title: KeyVaultAccessControlClient
 ```
 
 ### Tag: backuprestore
 These settings apply only when `--tag=backuprestore` is specified on the command line.
 
 ``` yaml $(tag) == 'backuprestore'
-input-file: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/backuprestore.json
+input-file: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.3/backuprestore.json
```
Comment (maorleger) [inline sdk/keyvault/azure-security-keyvault-administration/swagger/autorest.md:63]: I'd recommend using a specific commit here, ideally `e2ef44b87405b412403ccb005bfb3975411adf60` (same for any other autorest files)

File: sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## 4.4.0-beta.8 (Unreleased)
```
Comment (maorleger) [inline sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:3]: Should this be 4.4.0 or are you planning to remove the beta postfix in a follow-up commit?

**Review by maorleger — APPROVED** [review]
version changes LGTM, just a few comments but neither are blocking

File: sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md (hunk 2 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-keys/CHANGELOG.md:3]: I will update the CHANGELOG in a separate PR.

File: sdk/keyvault/azure-security-keyvault-administration/swagger/autorest.md (hunk 1 above)
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-administration/swagger/autorest.md:63]:
```suggestion
input-file: https://github.com/Azure/azure-rest-api-specs/blob/e2ef44b87405b412403ccb005bfb3975411adf60/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.3/backuprestore.json
```

File: sdk/keyvault/azure-security-keyvault-administration/swagger/autorest.md [hunk 3]
```diff
@@ -52,14 +52,14 @@ context-client-method-parameter: true
 These settings apply only when `--tag=rbac` is specified on the command line.
 
 ``` yaml $(tag) == 'rbac'
-input-file: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/keyvault/data-plane/Microsoft.KeyVault/preview/7.3-preview/rbac.json
+input-file: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.3/rbac.json
```
Comment (vcolin7) [inline sdk/keyvault/azure-security-keyvault-administration/swagger/autorest.md:55]:
```suggestion
input-file: https://github.com/Azure/azure-rest-api-specs/blob/e2ef44b87405b412403ccb005bfb3975411adf60/specification/keyvault/data-plane/Microsoft.KeyVault/stable/7.3/rbac.json
```

**Comment by vcolin7** [comment]
Overriding Check Enforcer as the pipeline is constantly failing due to some issues with Maven [only for a couple builds](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1473163&view=logs&j=595d1426-bc68-5e1a-c3c4-1388ac66f371&t=3f5614cb-a10e-5f36-56f9-bacc3805623f&l=75).

**Pull request merged**